src/settings/watchers/FontWatcher.ts

```diff
@@ -40,23 +41,26 @@ export class FontWatcher implements IWatcher {
 
     public constructor() {
         this.dispatcherRef = null;
-        this.migrateBaseFontSize();
     }
 
-    public start(): void {
+    public async start(): Promise<void> {
         this.updateFont();
         this.dispatcherRef = dis.register(this.onAction);
+        /**
+         * baseFontSize is an account level setting which is loaded after the initial
+         * sync. Hence why we can't do that in the `constructor`
+         */
+        await this.migrateBaseFontSize();
```

Comment:
start is called before sync data is loaded. It is called right after the constructor even.

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/3cf05877-3c00-4a0f-8850-559dc3b8dfba)

Reply:
Why is that an issue? If i'm looking at the current state of the codebase `baseFontSize` is only set on at the `SettingLevel.Device` level. We should not have to wait for the initial sync.

Or are you only saying that in regards to the code comment above? I will regardless change the comment so that it matches the reality

Reply:
Set yes. But it is read from account level. See https://github.com/matrix-org/matrix-react-sdk/pull/11103#discussion_r1243539733

Design switched it from account->device, there was no migration, so some users today will still be using their account-level setting unless they've changed font size yet again. Which means your migration would change their font scaling which would be jarring and not user friendly.